### PR TITLE
`Development`: Use signals in date time picker

### PR DESCRIPTION
**Pull request by florian-glombik**
#### General
- [x] I tested ~**all**~ the changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context
This PR introduces signals, and fixes code smells in components used for editing and updating lectures. This is a preparation for [Lectures: Add status bar to lecture creation and edit mode #9655](https://github.com/ls1intum/Artemis/pull/9655) and aims to keep #9655 at a manageable size of changes.

### Description
- Fixed code smells
- Introduced signals in `date-time-picker.component.ts` (and thereby removed method calls in the template)

### Steps for Testing
Prerequisites:
- 1 Instructor

THIS PR DOES NOT HAVE CODE REVIEWS YET, YOU MIGHT WANT TO WAIT WITH MANUAL TESTS UNTIL THE CODE REVIEWS ARE PRESENT

1. Navigate to the Instructor lecture view (creating and editing)
2. Create and edit a lecture, verify that the datepickers work the same as on develop
3. Create an exam and verify the datepickers work the same as on develop
4. Create an exercise (of one type, not all exercise types) (except for programming) and verify the datepickers work the same as on develop
5. If you know where one of the components below is in the UI, verify that the datepickeres works the same as on develop (take one component if possible with reasonable effort, you are not expected to test the whole list)

Please add your review comment on which component you have tested so other test reviews can consider it, and we can test different parts of the application.

- the datepicker is used throughout the following components
  - system-notification-management-update.component.html
  - common-course-competency-form.component.html
  - course-update.component.html
  - exam-update.component.html
  - file-upload-exercise-update.component.html
  - modeling-exercise-update.component.html
  - programming-assessment-repo-export-dialog.component.html
  - quiz-exercise-update.component.html
  - quiz-re-evaluate.component.html
  - participation.component.html
  - submission-export-dialog.component.html
  - text-exercise-update.component.html
  - lecture-attachments.component.html
  - attachment-unit-form.component.html
  - attachment-units.component.html
  - online-unit-form.component.html
  - text-unit-form.component.html
  - video-unit-form.component.html
  - lecture-update.component.html
  - lecture-wizard-period.component.html
  - build-queue.component.html
  - vcs-access-token-settings.component.html

This is a refactoring PR that does not add functionality, so the behavior should not change in comparison to the develop branch.

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots
UseSignalsInDateTimerPicker-9694

## Summary by CodeRabbit

- **New Features**
	- Enhanced date-time picker component with improved reactivity and dynamic updates.
	- Introduced new signals and computed properties for better state management.

- **Bug Fixes**
	- Updated validation logic for input fields to ensure accurate feedback on validity.
	- Improved test coverage by validating date calculations against expected outcomes.
	- Added tests for clearing the datepicker value to ensure proper functionality.

### CODE REVIEW

File: src/main/webapp/app/shared/date-time-picker/date-time-picker.component.ts [hunk 1]
```diff
@@ -16,26 +16,39 @@
...
-
-    readonly faCalendarAlt = faCalendarAlt;
-    readonly faGlobe = faGlobe;
-    readonly faClock = faClock;
-    readonly faQuestionCircle = faQuestionCircle;
-    readonly faCircleXmark = faCircleXmark;
-    readonly faTriangleExclamation = faTriangleExclamation;
+    labelName = input<string>();
+    labelTooltip = input<string>();
+    value = model<any>();
```
[the remark this answers is not in the document]
Comment (florian-glombik) [reply src/main/webapp/app/shared/date-time-picker/date-time-picker.component.ts:29]: Not sure if sometimes a string is passed in aswell, as the component is used throughout several places I will keep it like it was

**Review by edkaya — APPROVED** [review]
Code lgtm

**Review by JohannesWt — APPROVED** [review]
Code 🧑‍💻

**Review by PaRangger — APPROVED** [review]
Tested on TS4. Date Picker works fine and no related errors in the console. Code LGTM.

**Review by asliayk — APPROVED** [review]
code lgtm

**Review by HawKhiem — APPROVED** [review]
Tested on TS4. Works as expected

**Review by JerroyTan — APPROVED** [review]
Tested on TS4, Date picker functionalities worked fine.

**Pull request merged**